clients/ui/frontend/src/app/pages/modelCatalog/screens/RegisterCatalogModelForm.tsx

```diff
@@ -0,0 +1,306 @@
+    setData('versionName', initialFormData.versionName);
+    setData('versionDescription', initialFormData.versionDescription);
+    setData('sourceModelFormat', initialFormData.sourceModelFormat);
+    setData('sourceModelFormatVersion', initialFormData.sourceModelFormatVersion);
+    setData('modelLocationType', initialFormData.modelLocationType);
+    setData('modelLocationEndpoint', initialFormData.modelLocationEndpoint);
+    setData('modelLocationBucket', initialFormData.modelLocationBucket);
+    setData('modelLocationRegion', initialFormData.modelLocationRegion);
+    setData('modelLocationPath', initialFormData.modelLocationPath);
+    setData('modelLocationURI', initialFormData.modelLocationURI);
+    setData('modelRegistry', initialFormData.modelRegistry);
+    setData('modelCustomProperties', initialFormData.modelCustomProperties);
+    setData('versionCustomProperties', initialFormData.versionCustomProperties);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [model.id, preferredModelRegistry.name, versionTag]);
+
+  const [submittedRegisteredModelName, setSubmittedRegisteredModelName] =
+    React.useState<string>('');
+  const [submittedVersionName, setSubmittedVersionName] = React.useState<string>('');
+  const [registrationErrorType, setRegistrationErrorType] = React.useState<string | undefined>(
+    undefined,
+  );
+
+  const isModelNameValid = isNameValid(formData.modelName);
+  const isModelNameDuplicate = registeredModelsLoaded
+    ? isModelNameExisting(formData.modelName, registeredModels)
+    : false;
+  const hasModelNameError = !isModelNameValid || isModelNameDuplicate;
+
+  const isSubmitDisabled =
+    isSubmitting || isRegisterCatalogModelSubmitDisabled(formData, registeredModels);
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    setSubmitError(undefined);
+
+    // Additional validation before submission
+    if (!formData.sourceModelFormat || formData.sourceModelFormat.trim() === '') {
+      setSubmitError(new Error('Source model format is required'));
+      setIsSubmitting(false);
+      return;
+    }
+
+    if (!formData.sourceModelFormatVersion || formData.sourceModelFormatVersion.trim() === '') {
+      setSubmitError(new Error('Source model format version is required'));
+      setIsSubmitting(false);
+      return;
+    }
+
+    if (!formData.modelLocationURI || formData.modelLocationURI.trim() === '') {
+      setSubmitError(new Error('Model location URI is required'));
+      setIsSubmitting(false);
+      return;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (!apiState.api) {
+      setSubmitError(new Error('Model registry API is not available'));
+      setIsSubmitting(false);
+      return;
+    }
+
+    try {
+      const {
+        data: { registeredModel, modelVersion, modelArtifact },
+        errors,
+      } = await registerModel(apiState, formData, 'user'); // TODO: Get actual user
```

Comment:
You can get this value from the AppContext just like the regular RegisterModel form does.
```ts
  const { user } = React.useContext(AppContext);
  const author = user.userId || '';
```